packages/app/src/providers/mock/utils.ts

```diff
@@ -46,19 +43,22 @@ export function findMockEntity(
   return child;
 }
 
-export function sliceValue<T extends DType>(
-  value: unknown,
-  dataset: Dataset<ArrayShape | ScalarShape, T>,
+export function sliceValue(
+  value: unknown[],
+  dataset: Dataset<ArrayShape>,
   selection: string,
-): ScalarValue<T>[] {
+): unknown {
   const { shape } = dataset;
-  const dataArray = ndarray(value as ScalarValue<typeof dataset.type>[], shape);
-  const mappedArray = applyMapping(
```

Comment:
`applyMapping` brings some unnecessary misdirection. Instead, I now call `pick` and `createArrayFromView` directly.

And since `pick` always returns an array even when selecting a scalar, I add a condition to extract the scalar.